batavia/modules/_compile/tokenizer.js

```diff
@@ -0,0 +1,1095 @@
+      var quote = c;
+      var quote_size = 1;             // 1 or 3
+      var end_quote_size = 0;
+
+      // Find the quote size and start of string
+      c = tok.tok_nextc();
+      if (c == quote) {
+          c = tok.tok_nextc();
+          if (c == quote) {
+              quote_size = 3;
+          } else {
+              end_quote_size = 1;     // empty string found
+          }
+      }
+      if (c != quote) {
+          tok.tok_backup(c);
+      }
+
+      // Get rest of string
+      while (end_quote_size != quote_size) {
+          c = tok.tok_nextc();
+          if (c == EOF) {
+              if (quote_size == 3) {
+                  tok.done = E_EOFS;
+              } else {
+                  tok.done = E_EOLS;
+              }
+              tok.cur = tok.inp;
+              return [ERRORTOKEN, p_start, p_end, 11];
+          }
+          if (quote_size == 1 && c == '\n') {
+              tok.done = E_EOLS;
+              tok.cur = tok.inp;
+              return [ERRORTOKEN, p_start, p_end, 11];
+          }
+          if (c == quote) {
+              end_quote_size += 1;
+          } else {
+              end_quote_size = 0;
+              if (c == '\\') {
+                  c = tok.tok_nextc();  // skip escaped char
+              }
+          }
+      }
+
+      p_start = tok.start;
+      p_end = tok.cur;
+      return [STRING, p_start, p_end];
+  }
+  return batavia.builtins.None;
+};
+
+Tokenizer.prototype.fraction = function(c) {
+  var tok = this;
+  var e;
+  // Accept floating point numbers.
+  if (c == '.') {
+      // Fraction
+      do {
+          c = tok.tok_nextc();
+      } while (isdigit(c));
+  }
+  if (c == 'e' || c == 'E') {
+      return tok.exponent();
+  }
+  if (c == 'j' || c == 'J') {
+      /* Imaginary part */
+      c = tok.tok_nextc();
+  }
+
+  tok.tok_backup(c);
+  p_start = tok.start;
+  p_end = tok.cur;
+  return [NUMBER, p_start, p_end];
+};
+
+Tokenizer.prototype.exponent = function(c) {
+  var tok = this;
+  var e = c;
+  /* Exponent part */
+  c = tok.tok_nextc();
+  if (c == '+' || c == '-') {
+      c = tok.tok_nextc();
+      if (!isdigit(c)) {
+          tok.done = E_TOKEN;
+          tok.tok_backup(c);
+          return [ERRORTOKEN, tok.start, tok.end, 15];
+      }
+  } else if (!isdigit(c)) {
+      tok.tok_backup(c);
+      tok.tok_backup(e);
+      return [NUMBER, tok.start, tok.end];
+  }
+  do {
+      c = tok.tok_nextc();
+  } while (isdigit(c));
+
+  if (c == 'j' || c == 'J') {
+      return tok.imaginary();
+  }
+
+  tok.tok_backup(c);
+  return [NUMBER, tok.start, tok.cur];
+};
+
+Tokenizer.prototype.imaginary = function() {
+  var tok = this;
+  var c = tok.tok_nextc();
+  tok.tok_backup(c);
+  p_start = tok.start;
+  p_end = tok.cur;
+  return [NUMBER, p_start, p_end];
+};
+
+
+Tokenizer.prototype.tok_nextc = function() {
+var tok = this;
+
+if (tok.cur != tok.inp) {
+    return tok.buf[tok.cur++]; /* Fast path */
+}
+return EOF;
+
+// for (;;) {
```

Comment:
You're right. I don't think we need this block of code anymore. It is intended if we are reading the file chunks at a time, like from a disk, but we assume we have it as a string already. I'll delete it.